app/controllers/email_controller.rb

```diff
@@ -5,17 +5,18 @@ class EmailController < ApplicationController
   ZENDESK_SMS_REGEX = /〒\nphone: \+(?<phone_number>[0-9]{11})\nticket_id: (?<ticket_id>[0-9]*)\nbody: (?<body>.+)\n〶/m
 
   def create
-    raise StandardError unless sent_to_known_email_hook?
-
-    body = params[:text]
-    match = body.match(ZENDESK_SMS_REGEX)
+    unless sent_to_known_email_hook?
+      raise StandardError.new("Email webhook received invalid email to #{params[:to]}")
+    end
 
-    raise "Could not parse incoming message (#{params[:subject]}) from: #{params[:from]}" unless match
+    unless valid_update_from_user? || from_agent_replying_to_thread?
```

Comment:
the conditionals here confused me a bit, specifically this one in comparison to the next conditional (switching between `unless` and `if`). I think I personally find the combination more readable if this one uses the `if` variant 

> if !valid_update_from_user? && !from_agent_replying_to_thread?

this helps make the logic more clear the cases that cause the error vs the cases that cause the logic below to happen.

Reply:
Thank you for the input, it's a good point. I agree that the conditionals are a bit confusing. I'm going to leave it because:
1. I feel like both types of conditionals are a bit confusing and slightly prefer the `unless`. But happy to talk more to reach a consensus on situations like this.
2. I want the whole thing to go away and become a webhook!

Reply:
Makes sense, yeah! I think there is some unavoidable confusion since there's tricky logic to capture. Thanks for considering!